dd-sdk-android-internal/src/main/java/com/datadog/android/internal/profiler/DDExecutionTimer.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.internal.profiler
+
+internal class DDExecutionTimer(
+    private val benchmarkSdkPerformance: BenchmarkSdkPerformance = GlobalBenchmark.getSdkPerformance()
+) : ExecutionTimer {
+    private var track: String = ""
+
+    override fun setTrackName(track: String) {
+        this.track = track
+    }
```

Comment:
why do we allow switching tracks here instead of creating timer per track?

Reply:
There's actually only one track per timer - GlobalBenchmark determines whether to return the actual timer or a noop - one instance per feature, and this is then setup with the appropriate track name using this method.

Reply:
I think we still can get rid of this setter and move it to the constructor argument